spring-cloud-gcp-data-spanner/src/test/java/org/springframework/cloud/gcp/data/spanner/test/SkipWhenNoSpanner.java

```diff
@@ -0,0 +1,69 @@
+/*
+ *  Copyright 2018 original author or authors.
+ *
+ *  Licensed under the Apache License, Version 2.0 (the "License");
+ *  you may not use this file except in compliance with the License.
+ *  You may obtain a copy of the License at
+ *
+ *       http://www.apache.org/licenses/LICENSE-2.0
+ *
+ *  Unless required by applicable law or agreed to in writing, software
+ *  distributed under the License is distributed on an "AS IS" BASIS,
+ *  WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+
+package org.springframework.cloud.gcp.data.spanner.test;
+
+import com.google.cloud.spanner.DatabaseAdminClient;
+import org.junit.AssumptionViolatedException;
+import org.junit.rules.TestRule;
+import org.junit.runner.Description;
+import org.junit.runners.model.Statement;
+
+/**
+ * @author Balint Pato
+ */
+public class SkipWhenNoSpanner implements TestRule {
+
+	private final DatabaseAdminClient databaseAdminClient;
+
+	private final String instanceId;
+
+	SkipWhenNoSpanner(DatabaseAdminClient databaseAdminClient, String instanceId) {
+		this.databaseAdminClient = databaseAdminClient;
+		this.instanceId = instanceId;
+	}
+
+	@Override
+	public Statement apply(Statement base, Description description) {
+		DatabaseAdminClient databaseAdminClient = this.databaseAdminClient; // for checkstyle...
```

Comment:
What's up with checkstyle? It doesn't let you declare these inside the lambda? Then should we get rid of the comments?

Reply:
checkstyle forces the use `this` for fields, this is an issue inside the anonymous class (I wish it could be a lambda, then it would work, but JUnit Statement is an abstract class) as `this` refers to the instance of the anonymous class itself, not the JUnit test object.